src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/rmd/display/DefaultChunkOptionsPopupPanel.java

```diff
@@ -224,14 +308,152 @@ else if (modeId == "mode/r")
             endIndex = cursor.getIndex();
          }
 
+         String optionName = StringUtil.substring(arguments, startIndex, equalsIndex).trim();
          chunkOptions.put(
-               StringUtil.substring(arguments, startIndex, equalsIndex).trim(),
-               StringUtil.substring(arguments, equalsIndex + 1, endIndex).trim());
+               ChunkOptionValue.normalizeOptionName(optionName),
+               new ChunkOptionValue(
+                  StringUtil.substring(arguments, equalsIndex + 1, endIndex).trim(),
+                  OptionLocation.FirstLine));
 
          startIndex = cursor.getIndex() + 1;
       }
       while (cursor.moveToNextCharacter());
    }
 
+   /**
+    * Returns list of option lines from the editor; these follow the first line,
+    * must begin with "#| ", and must have at least one non-whitespace character
+    * after "#| ". The lines aren't checked to see if they contain valid YAML, and
+    * the option names are not normalized (e.g. fig.cap vs. fig-cap).
+    * 
+    * @param startLine first potential option line
+    * @return list of option lines without the leading #|
+    */
+   private List<String> getYamlOptionLines(int startLine)
+   {
+      List<String> optionLines = new ArrayList<>();
+      int currentLine = startLine;
+      String line;
+      while ((line = display_.getLine(currentLine)) != null)
+      {
+         // a line with no characters after the starting #| isn't valid and ends detection
+         if (line.startsWith("#| ") && line.trim().length() > 2)
```

Comment:
Ultra-nit: if trailing whitespace is included, then content following that will be parsed by knitr. For example:

`````r
```{r}
#| example: true
#|           
#| example2: false
print(knitr::opts_current$get("example"))
print(knitr::opts_current$get("example2"))
```
`````

In this case, both `example` and `example2` will be set. Hopefully people aren't actually relying on this in the wild though; I imagine this is kind of "unspecified" behavior.